source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,162 @@
+either succeed together or fail together when one or more of the
+operations fail.
+
+In MongoDB, transactions run within logical **sessions**. A session is a
+grouping of related read or write operations that you want to run
+sequentially. Sessions enable :manual:`causal consistency </core/read-isolation-consistency-recency/#causal-consistency>`
+for a group of operations or allow you to execute operations in an :website:`ACID
+transaction </basics/acid-transactions>`. MongoDB guarantees that the data
+involved in your transaction operations remains consistent, even if the operations
+encounter unexpected errors.
+
+With the {+driver-short+}, you can create a new session from a
+``Client`` instance as a ``Session`` type. You should only use a
+``Session`` instance created from a client to execute operations
+using that client instance or in a database or collection selected from that
+client.
+
+.. warning::
+   
+   You should not use custom implementations of the ``Session``
+   interface in production. Additionally, implementations of ``Session``
+   are not safe for concurrent use by multiple `goroutines
+   <https://www.golang-book.com/books/intro/10>`__.
+
+Methods
+-------
+
+After you start a session using the ``StartSession()`` method, you can modify
+the session state using the method set in the ``Session`` interface. The
+following table describes these methods:
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``StartTransaction()``
+     - | Starts a new transaction, configured with the given options, on
+         this session. Returns an error if there is already
+         a transaction in progress for the session. For more
+         information, see the :manual:`Manual entry </reference/method/Session.startTransaction/>`.
+       |
+       | **Parameter**: ``TransactionOptions``
+       | **Return Type**: ``error``
+
+   * - ``AbortTransaction()``
+     - | Aborts the active transaction for this session. Returns an
+         error if there is no active transaction for the session or the
+         transaction has been committed or aborted. For more
+         information, see the :manual:`Manual entry </reference/method/Session.abortTransaction/>`.
+       |
+       | **Parameter**: ``Context``
+       | **Return Type**: ``error``
+
+   * - ``CommitTransaction()``
+     - | Commits the active transaction for this session. Returns an
+         error if there is no active transaction for the session or the
+         transaction has been aborted. For more
+         information, see the :manual:`Manual entry </reference/method/Session.commitTransaction/>`.
+       |
+       | **Parameter**: ``Context``
+       | **Return Type**: ``error``
+
+   * - ``WithTransaction()``
+     - | Starts a transaction on this session and runs the ``fn``
+         callback. For more
+         information, see the :manual:`Manual entry </reference/method/Session.withTransaction/>`.
```

Comment:
For the most part, I linked the Manual docs because those pages provide more background on each of the methods. I can remove all of them if you think the mismatch between shell syntax and Go syntax will cause confusion.

Reply:
The rest of the MongoDB manual links go to reference information about the server APIs that correspond to the `Session` API functions, which makes sense. However, since `WithTransaction` is a driver/`mongosh`-only convenience function, there is no corresponding server API. TBH I'm not sure why there is a MongoDB manual page for `WithTransaction` at all because it confusingly mixes server APIs with `mongosh` APIs.

I think removing the link to the MongoDB manual for `WithTransaction` is a good idea because it's otherwise confusing.